docs/en/04_Changelogs/4.0.0.md

```diff
@@ -28,12 +85,187 @@ Note that this will not allow you to utilise certain file versioning features in
+is applied to both `File` and `DBFile`. These both implement a common interface `AssetContainer`, which
+has the `getIsImage()` method. In some cases, it may be preferable to invoke this method to detect
+if the asset is an image or not, rather than checking the subclass, as the asset may also be a `DBFile` with
+an image filter applied, rather than an instance of the `Image` dataobject.
+
+In addition, a new file category `image/supported` has been added, which is a subset of the `image` category.
+This is the subset of all image types which may be assigned to the `[api:Image]` dataobject, and may have
+manipulations applied to it. This should be used as the file type restriction on any `[api:UploadField]` which
+is intended to upload images for manipulation. All image categories have also been specified via constants
+on the `[api:File]` DataObject.
+
+Before:
+
+
+	:::php
+	if($file instanceof Image) {
+		$upload = new UploadField();
+		$upload->setAllowedFileCategories('image');
+	}
+
+
+After:
+
+
+	:::php
+	if($file->getIsImage()) {
+		$upload = new UploadField();
+		$upload->setAllowedFileCategories(File::CATEGORY_IMAGE_SUPPORTED);
+	}
+
+
+In cases where image-only assets may be assigned to relationships then your datamodel should specify explicitly
+an `Image` datatype, or refer to `DBFile('image/supported')`.
+
+E.g.
+
+
+	:::php
+	class MyObject extends DataObject {
+		private static $has_one = array(
+			"ImageObject" => "Image"
+		);
+		private static $db = array(
+			"ImageField" => "DBFile('image/supported')"
+		);
+	}
+
+
+### Upgrading code that writes to `File` dataobjects, or writes files to the 'assets' folder
+
+In the past all that was necessary to write a `File` dataobject to the database was to ensure a physical file
+existed in the assets folder, and that the Filename of the dataobject was set to the same location.
+
+Since the storage of physical files is no longer a standard location, it's necessary to delegate the writing of such
+files to the asset persistence layer. As a wrapper for an individual file, you can use any of the `setFrom*`
+methods to assign content from a local (e.g. temporary) file, a stream, or a string of content.
+
+You would need to upgrade your code as below.
+
+Before:
+
+
+	:::php
+	function importTempFile($tmp) {
+		copy($tmp, ASSETS_PATH . '/imported/' . basename($tmp));
+		$file = new File();
+		$file->setFilename('assets/imported/'.basename($tmp));
+		$file->write();
+	}
+
+
+After:
+
+
+	:::php
+	function importTempFile($tmp) {
+		$file = new File();
+		$file->setFromLocalFile($tmp, 'imported/'.basename($tmp));
+		$file->write();
+	}
+
+
+Note that 'assets' is no longer present in the new code, and the path beneath what was once assets is now
+used to generate the 'filename' value. This is because there is no longer an assumption that files are
+stored in the assets folder.
+
+Since the structure of `File` dataobjects has changed, a new task `MigrateFileTask` has been added to assist
```

Comment:
This should probably get a more prominent position in the changelog overview, to avoid devs just running "dev/build" (which is usually assumed to be fairly non-destructive). Also, `FileMigrationHelper` looks like it might run out of memory or exeuction time on large filesystems - we should add a recommendation to run this from the command-line, and use `increase_time_limit_to()`/`increase_memory_limit_to()` the in `MigrateFileTask`